pkg/registry/credentials.go

```diff
@@ -0,0 +1,52 @@
+package registry
+
+import (
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/docker/cli/cli/config/credentials"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// CredentialStore for registry credentials such as ~/.docker/config.json.
+type CredentialStore struct {
+	directory  string
+	configFile *configfile.ConfigFile
+}
+
+// NewCredentialStore create a create store.
+func NewCredentialStore(args ...string) CredentialStore {
```

Comment:
nit: doc comment about how to use args to define a directory?

Reply:
I just added a SetDirectory method instead.